### ISSUE TITLE
Upgrade to 0.56.0 caused several network errors

### ISSUE THREAD

**Issue by jraede** [rancher/rancher#3430]
We're currently running:
```
Rancher	v0.51.0
Cattle	v0.130.0
User Interface	v0.78.0
Rancher Compose	v0.7.0
```

I upgraded to 0.56.0 earlier, and it caused the following problems:

* service links across physical servers stopped working (ending up with request timeouts). Restarting the network agent, restarting the services, and restarting the physical server did not fix the issue. Services linked to services on the *same* physical server, however, did not have problems
* Probably related - Load Balancer services went from "active" (green) to "initializing" (yellow) and remained there through restarts, etc.
* After trying several different things, we ended up fixing the problem by reverting to rancher-server 0.51.0 and that immediately fixed all networking problems.

Were there significant changes to the service discovery/networking that may have caused this in 0.56? Any other ideas for how this could have been fixed?

Thanks

**Comment by deniseschannon** [comment]
Yes, there were significant changes in networking (switching from racoon to charon) as indicated in our release notes. https://github.com/rancher/rancher/releases/tag/v0.56.0

What OS are you using? If you are using CoreOS, you'll need to upgrade to v0.56.1 as we discovered that there were other issues in CoreOS that we quickly addressed.

If you are running another OS, please let me know so I can try to investigate further for you.

**Comment by jraede** [comment]
Our rancher admin server is running RancherOS and the servers are all Ubuntu 14.04

**Comment by deniseschannon** [comment]
When you say "restarting the physical server", did you restart all the Ubuntu servers? 

All of your issues are due to cross host communication failing.

**Comment by jraede** [comment]
Correct. We did the following in order:

* Restart the services (both sides of the link)
* Restart the network agent on all servers
* Restarting the physical ubuntu server via the EC2 dashboard (although not *all* servers, since we had some services that were unaffected, so we shouldn't necessarily rule this out as a solution)

**Comment by deniseschannon** [comment]
Would you be willing to try from v0.51.0 to v0.56.1? I found out that there was also a fix where host reboot/network restart did not restore connectivity.

**Comment by jraede** [comment]
Sure, it'll have to be later tonight though since we have some people relying on the services now so I need to wait until off-hours. I'll let you know the result.

**Comment by deniseschannon** [comment]
Here's some additional info that would help. Exec into the network agent and do a `ps-eaf | grep rancher-net`.  You should get something like below. Ideally, for ubuntu, centos, rancheros, it should  have `--gcm=true` and for coreos, it should have `--gcm=false`. I know you're running ubuntu for your hosts, but wanted to share the information for anyone else who looks at the issue. 

```
root      3659     1  0 Jan26 ?        00:01:57 /var/lib/cattle/bin/rancher-net 
--log /var/log/rancher-net.log -f /var/lib/cattle/etc/cattle/ipsec/config.json -
c /var/lib/cattle/etc/cattle/ipsec -i 172.17.0.2/16 --pid-file /var/run/rancher-
net.pid --gcm=true                                                              
root     18025 18021  0 20:10 pts/0    00:00:00 grep rancher-net             
```

**Comment by jraede** [comment]
@deniseschannon here's the result of that; not sure what I should be looking for. Do not see `--gcm` either way.

```
root     21913     1  0 16:19 ?        00:00:21 /var/lib/cattle/bin/rancher-net 
--log /var/log/rancher-net.log -f /var/lib/cattle/etc/cattle/ipsec/config.json -
c /var/lib/cattle/etc/cattle/ipsec -i <ip>/16 --pid-file /var/run/rancher-
net.pid  
```

**Comment by jraede** [comment]
@deniseschannon upgrading to 0.56.1 + restarting all network agents did not resolve the issue. Would you advise restarting all physical hosts?

**Comment by deniseschannon** [comment]
Yes, that's the next step. If that doesn't recover it, we'll have to look further into it. Can you try that?

**Comment by jraede** [comment]
I need to do a bit of coordinating but yes I will try that within the next few days and let you know the results.

**Comment by jraede** [comment]
@deniseschannon looks like I'm experiencing the same issue as in https://github.com/rancher/rancher/issues/3436 - should the output of `ps -eaf | grep "rancher-net"` show up with `--gcm=true` even while running 0.51.0? Or is that a new flag with 0.56?

**Comment by deniseschannon** [comment]
Based on the comments on this forum post, you could also try deleting the network agent and having Rancher re-create them. That's what fixed someone else who was running ubuntu and having network issues. 

https://forums.rancher.com/t/upgraded-to-v0-56-and-hosts-are-getting-local-addresses/1556/6

**Comment by jraede** [comment]
Does Rancher automatically recreate them on delete?

**Comment by deniseschannon** [comment]
Yes, Rancher will automatically create them.

**Comment by wundram** [comment]
We are seeing the issue since upgrading to 56.1 where our load balancers get stuck in "initializing" after deployment.   At least in some cases they are functional, but always show as yellow, which makes the stack show as degraded.

Did anyone have any success clearing this by rebooting hosts or network agents?

**Comment by jraede** [comment]
@wundram @deniseschannon  we're setting up a production-only rancher environment as part of our sprint next week so we can try out rancher upgrades for dev/staging without affecting it. Once we do that then I'll be able to play around in our dev environment to try to reproduce the issue and see what combination of restarting/deleting containers and servers ends up resolving it. Will report back here.

**Comment by deniseschannon** [comment]
@wundram See https://github.com/rancher/rancher/issues/3413

In that issue, most people have gotten their network connectivity working (restart network agent, restart host, delete network agent) which will solve the LB initializing issue.

**Comment by jraede** [comment]
I was able to reproduce the above issue in a dev environment and resolve it by upgrading to 0.56.1 and deleting network agents, so I will be trying that in the original problem environment later this week and will confirm here.

**Comment by jraede** [comment]
Unfortunately deleting + recreating the network agents did not fix this issue in the original environment this happened on. I think this must be because we started that environment at 0.44.0 and the one that I was able to reproduce it and fix it successfully was started directly at 0.51.0. In any case, I just ended up replacing the entire environment by spinning up new servers, so I can't verify this either way. Up to you guys if you want to close.
